app/javascript/courses/review/components/CoursesReview__ChecklistEditor.res

```diff
@@ -226,19 +307,41 @@ let make = (~reviewChecklist, ~updateReviewChecklistCB, ~closeEditModeCB, ~targe
                                   )}
                               />
                               <div
-                                className="flex w-10 h-10 absolute top-0 right-0 mr-1 items-center justify-center">
-                                <button
-                                  title={t("checklist_item_title.remove_button_title")}
-                                  className="flex items-center justify-center bg-gray-100 w-7 h-7 mt-px text-sm text-gray-700 hover:text-red-600 hover:bg-red-100 rounded-full ml-2 border border-transparent text-center"
-                                  onClick={_ =>
-                                    removeChecklistResult(
-                                      itemIndex,
-                                      resultIndex,
-                                      reviewChecklistItem,
-                                      setState,
-                                    )}>
-                                  <Icon className="if i-times-regular" />
-                                </button>
+                                className="flex h-10 absolute top-0 right-0 mr-1 items-center justify-center">
+                                {controlIcon(
+                                  ~icon="fa-arrow-up",
+                                  ~title={t("checklist_item_title.move_up_button_title")},
+                                  ~handler=moveChecklistResultUp(
+                                    itemIndex,
+                                    resultIndex,
+                                    reviewChecklistItem,
+                                    setState,
+                                  ) |> OptionUtils.map((cb, _) => cb()),
+                                )}
```

Comment:
Why should we have a complex implementation for this?
Try this out
```suggestion
                                {controlIcon(
                                  ~icon="fa-arrow-up",
                                  ~title={t("checklist_item_title.move_up_button_title")},
                                  ~handler={
                                    _ =>
                                      updateChecklistItem(
                                        ReviewChecklistItem.moveResultItemUp(
                                          resultIndex,
                                          reviewChecklistItem,
                                        ),
                                        itemIndex,
                                        setState,
                                      )
                                  },
                                  ~hidden={resultIndex <= 0},
                                )}
```

And `controlIcon` can simply be 
```res
let controlIcon = (~icon, ~title, ~handler, ~hidden) =>
  ReactUtils.nullIf(
    <button
      title
      disabled={hidden}
      className="px-2 py-1 focus:outline-none text-sm text-gray-700 hover:bg-gray-300 hover:text-gray-900 overflow-hidden"
      onClick={handler}>
      <i className={"fas fa-fw " ++ icon} />
    </button>,
    hidden,
  )
```